### PR TITLE
ksmbd-tools: update to 3.4.6

### PR DESCRIPTION
**Pull request by neheb**
Switch to git tarball as the meson files did not get added to the official one.

Backport busybox style binaries. Saves on size.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: nobody

### CODE REVIEW

**Comment by Gingernut1978** [comment]
[ksmbd.adduser/7870]: ERROR: Use of `/etc/ksmbd/smb.conf' is deprecated, rename it to `/etc/ksmbd/ksmbd.conf' now!

Looks like smb.conf was renamed to ksmbd.conf

**Comment by neheb** [comment]
That would probably require wsdd2 changes too though...

**Comment by neheb** [comment]
@Gingernut1978 done with the changes. totally untested.

File: net/ksmbd-tools/files/ksmbd.init [hunk 1]
```diff
@@ -38,7 +38,7 @@ smb_header()
 	    -e "s#|WORKGROUP|#$workgroup#g" \
 	    -e "s#|DESCRIPTION|#$description#g" \
 	    -e "s#|INTERFACES|#$interfaces#g" \
-	    /etc/ksmbd/smb.conf.template > /var/etc/ksmbd/smb.conf
+	    /etc/ksmbd/ksmbb.conf.template > /var/etc/ksmbd/ksmbd.conf
```
Comment (zdz2019) [inline net/ksmbd-tools/files/ksmbd.init:41]: typo. "ksmbb"

Comment (neheb) [reply net/ksmbd-tools/files/ksmbd.init]: Fixed.

**Comment by neheb** [comment]
these hard reset changes in the init script are weird. Anyone know if they're still needed?

**Comment by Gingernut1978** [comment]
@neheb so far working fine but there is some log spam

```Sat Nov  5 07:26:58 2022 kern.err kernel: [86060.666940] ksmbd: v5 version is not supported
Sat Nov  5 07:26:58 2022 kern.err kernel: [86060.671422] ksmbd: v5 version is not supported
Sat Nov  5 07:26:58 2022 kern.err kernel: [86060.675881] ksmbd: v5 version is not supported
Sat Nov  5 07:26:58 2022 kern.err kernel: [86060.680339] ksmbd: v5 version is not supported
Sat Nov  5 07:26:58 2022 kern.err kernel: [86060.684833] ksmbd: v5 version is not supported
Sat Nov  5 07:26:58 2022 kern.err kernel: [86060.689291] ksmbd: v5 version is not supported
Sat Nov  5 07:26:58 2022 kern.err kernel: [86060.693772] ksmbd: v5 version is not supported
Sat Nov  5 07:26:58 2022 kern.err kernel: [86060.698231] ksmbd: v5 version is not supported
Sat Nov  5 07:27:01 2022 kern.err kernel: [86063.734944] ksmbd: v5 version is not supported
Sat Nov  5 07:27:03 2022 kern.err kernel: [86065.461679] ksmbd: v5 version is not supported
Sat Nov  5 07:27:03 2022 kern.err kernel: [86065.479949] ksmbd: v5 version is not supported
Sat Nov  5 07:27:03 2022 kern.err kernel: [86065.495279] ksmbd: v5 version is not supported
Sat Nov  5 07:27:03 2022 kern.err kernel: [86065.520714] ksmbd: v5 version is not supported
Sat Nov  5 07:27:03 2022 kern.err kernel: [86065.525192] ksmbd: v5 version is not supported
Sat Nov  5 07:27:03 2022 kern.err kernel: [86065.529654] ksmbd: v5 version is not supported
Sat Nov  5 07:27:03 2022 kern.err kernel: [86065.534155] ksmbd: v5 version is not supported

**Comment by neheb** [comment]
That's from the kernel module. I'll look into it.

**Comment by neheb** [comment]
@Gingernut1978 is this from an NTFS volume? the error has to do with DOS attributes.

**Comment by neheb** [comment]
looking closer at the code I see that this is for smb1. smb1 support will go away once everything migrates to kernel 5.15. In any case, I would report this upstream at https://github.com/cifsd-team/ksmbd

**Comment by Gingernut1978** [comment]
@neheb no, this is a ext4 drive. I'll check out reporting it upstream.

Not sure if you also maintain the luci-app-ksmbd but a few changes are needed to point to the new ksmbd.conf file.

Thanks

**Comment by neheb** [comment]
I do not. The old maintainer is no longer involved with OpenWrt.

**Comment by Gingernut1978** [comment]
> @Gingernut1978 is this from an NTFS volume? the error has to do with DOS attributes.

Seems you are correct, adding the **store dos attributes = no** variable to global config stops the log spam warning.

**Comment by ptpt52** [comment]
the 010, 020 patches could be drop is we use the latest code in https://github.com/cifsd-team/ksmbd-tools/commits/master
@neheb

**Comment by neheb** [comment]
Right. I'd rather use released ones though.

**Comment by Neustradamus** [comment]
@neheb: Have you seen this ticket?
- https://github.com/namjaejeon/ksmbd/issues/413

**Pull request merged**